src/Framework/ContaoFramework.php

```diff
@@ -0,0 +1,478 @@
+use Contao\CoreBundle\Exception\AjaxRedirectResponseException;
+use Contao\CoreBundle\Exception\IncompleteInstallationException;
+use Contao\CoreBundle\Exception\InvalidRequestTokenException;
+use Contao\Input;
+use Contao\System;
+use Symfony\Component\DependencyInjection\ContainerAwareTrait;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\RequestStack;
+use Symfony\Component\HttpFoundation\Session\SessionInterface;
+use Symfony\Component\Routing\RouterInterface;
+use Symfony\Component\Security\Csrf\CsrfToken;
+use Symfony\Component\Security\Csrf\CsrfTokenManagerInterface;
+
+/**
+ * Initializes the Contao framework.
+ *
+ * @author Christian Schiffler <https://github.com/discordier>
+ * @author Yanick Witschi <https://github.com/toflar>
+ * @author Leo Feyer <https://github.com/leofeyer>
+ * @author Dominik Tomasi <https://github.com/dtomasi>
+ * @author Andreas Schempp <https://github.com/aschempp>
+ */
+class ContaoFramework implements ContaoFrameworkInterface
+{
+    use ContainerAwareTrait;
+
+    /**
+     * @var RouterInterface
+     */
+    private $router;
+
+    /**
+     * @var CsrfTokenManagerInterface
+     */
+    private $tokenManager;
+
+    /**
+     * @var SessionInterface
+     */
+    private $session;
+
+    /**
+     * @var string
+     */
+    private $rootDir;
+
+    /**
+     * @var AdapterInterface
+     */
+    private $config;
+
+    /**
+     * @var string
+     */
+    private $csrfTokenName;
+
+    /**
+     * @var int
+     */
+    private $errorLevel;
+
+    /**
+     * @var RequestStack
+     */
+    private $requestStack;
+
+    /**
+     * @var Request
+     */
+    private $request;
+
+    /**
+     * @var bool
+     */
+    private static $initialized = false;
+
+    /**
+     * @var array
+     */
+    private $basicClasses = [
+        'System',
+        'Config',
+        'ClassLoader',
+        'TemplateLoader',
+        'ModuleLoader',
+    ];
+
+    /**
+     * Adapter class cache
+     * @var array
+     */
+    private $adapterCache = [];
+
+    /**
+     * Constructor.
+     *
+     * @param RequestStack              $requestStack  The request stack
+     * @param RouterInterface           $router        The router service
+     * @param SessionInterface          $session       The session service
+     * @param string                    $rootDir       The kernel root directory
+     * @param CsrfTokenManagerInterface $tokenManager  The token manager service
+     * @param string                    $csrfTokenName The name of the token
+     * @param AdapterInterface          $config        The config adapter object
+     * @param int                       $errorLevel    The PHP error level
+     */
+    public function __construct(
+        RequestStack $requestStack,
+        RouterInterface $router,
+        SessionInterface $session,
+        $rootDir,
+        CsrfTokenManagerInterface $tokenManager,
+        $csrfTokenName,
+        $config = null,
```

Comment:
I will, it shouldn't be allowed to override the $config anyway as this would cause `$this->config->getInstance()` to fail if somebody still passes the deprecated `ConfigAdapter`. I'm trying to fix this but there's a phpunit bug which prevents me from fixing it atm :D